assignments/HW5.md

```diff
@@ -0,0 +1,119 @@
+
+The deliverable for this part, at a high level, is a Design Document/RFC that explains and justifies the tooling you propose to incorporate into your process, and how you propose to do so. This decision should be feature- and data-driven, and should consider usability and process questions like how and when the tooling will be applied, and by whom. See below for more details.
+
+**NOTE: you do not need to integrate ALL `N` tools into your repo, but you should integrate at least (1) tool so that it runs via CI, and have one commit on which it was run.  You should reference this commit in your design document**
+
+#### Starter list of Tools
+
+Teedy has a Java backend and an AngularJS (JavaScript) frontend. Below are non-exhaustive lists of analysis tools that are available for both Java and JavaScript. Others are available. [Awesome Static Analysis page/repo](https://github.com/analysis-tools-dev/static-analysis) and [Awesome Dynamic Analysis page/repo](https://github.com/analysis-tools-dev/dynamic-analysis/) have extensive listings of available static and dynamic analysis tools for a pretty hefty list of programming languages, including Java. Some of the tools have GitHub Actions workflows on [GitHub Marketplace](https://github.com/marketplace?category=&query=&type=actions&verification=); use your Googling skills, and see what you find! 
+
+**Java**
+
+Static analysis:
+- [Checkstyle](https://github.com/checkstyle/checkstyle), a tool for checking Java source code for adherence to a code standard.
+- [SpotBugs](https://spotbugs.github.io/), a fork of [FindBugs](https://findbugs.sourceforge.net/) that can spot [over 400 bug patterns](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html)
+
+Dynamic analysis:
+- [Pitest](https://pitest.org/), a [mutation testing](https://en.wikipedia.org/wiki/Mutation_testing) system for Java.
+- [YourKit](https://www.yourkit.com/java/profiler/features/), commercial Java profiler (15 days trial)
+
+**JavaScript**
+
+Static analysis:
+- [eslint](https://eslint.org/) is a widely used linter in modern JavaScript packages. 
+- [flow](https://flow.org/) is a static type checker for JavaScript.
+
+Dynamic analysis:
+- [jalangi2](https://github.com/Samsung/jalangi2]) is a dynamic analysis framework for JavaScript. 
```

Comment:
Link is broken. I think there is an extra `]` in the URL.

Reply:
Good catch. Thanks!